### ISSUE TITLE
Record Computer Audio

### ISSUE THREAD

**Issue by KrishnanS2006** [EasyScreenCast/EasyScreenCast#307]
I have been using EasyScreenCast for a while now, but I have never been able to record the audio of my screen. If, for example, I am recording a meeting using the tool, my own voice is recorded, but never the actual audio coming from the meeting. Is this a setting I am missing, a bug with EasyScreenCast, or a feature yet to be implemented?

## Expected Behavior
Audio from my mic and audio from the computer are both included in the final video.

## Current Behavior
Only the audio from my mic is saved to the recording.

## Your Environment

* Where did you download the extension?
 [x] Gnome shell extension website
 [ ] From Git (Add commit tag)
* Gnome shell version: 3.36.9

* Operating System and version: Ubuntu 20.04.2

**Comment by adangel** [comment]
You can select the audio device which is recorded together with the screencast. EasyScreenCast not only shows the microphones but also the monitor devices. So you can e.g. record a screencast of a video including the sound.
But for video conferences, probably two audio devices need to be recorded: your own microphone and what you are hearing (unless the video conference software would play back your own microphone audio, which is unlikely due to latency).

So, maybe EasyScreenCast needs an option, to select multiple audio devices?

**Comment by adangel** [comment]
It seems to be possible with the following custom gstreamer pipeline:

```
queue max-size-buffers=0 max-size-time=0 max-size-bytes=0
! vp8enc min_quantizer=4 max_quantizer=13 cpu-used=2 deadline=500000 sharpness=0 target-bitrate=15000 threads=%T
! mux.

pulsesrc device="alsa_input.usb-046d_0825_816ECE10-02.mono-fallback"
! audioconvert
! queue max-size-buffers=0 max-size-time=0 max-size-bytes=0
! liveadder name=m_adder
! vorbisenc
! mux.

pulsesrc device="alsa_output.usb-DisplayLink_Dell_D3100_USB3.0_Dock_1512280507-02.analog-stereo.monitor"
! audioconvert
! queue max-size-buffers=0 max-size-time=0 max-size-bytes=0
! m_adder.

webmmux name=mux
```

You'll need to adjust the pulse-audio source devices, of course. But this pipeline mixes together two audio sources with live adder and puts the video stream and a single audiostream into the webm container.

**Comment by KrishnanS2006** [comment]
Yep - I tried this and it works! Thanks a lot! Although, it would be much nicer for EasyScreenCast to have built-in support for this, much like the multiple audio sources feature you mentioned. possibly a checklist and not a select dropdown?

**Comment by JonBlackwell** [comment]
I second this feature... maybe the option to choose a speaker/monitor and another option to choose a mic.

**Comment by johnstonesnow** [comment]
I am posting here rather than open an issue myself. I have just tried out this extension as someone recommended it for a screen recording with audio. Trouble is, I can only record audio from microphones, I was hoping to record 'system audio', i.e. internally rather than through a mic on my desk, so the user and can hear what I am hearing from my machine, rather than me and the output of my machine's speakers. 
I would love this option, is there any chance it may be added as a user-selectable option in settings some time in the future?
thanks, great extension otherwise
